frontend/src/helpers/index.ts

```diff
@@ -15,7 +17,22 @@ export const formatNameOrId = (node?: MaybeNamed, idPrefix: string = ''): string
     if (!node) {
         return '';
     }
-    return node.name?.value ? node.name.value : `${idPrefix}${formatShortId(node.id)}`;
+    if (node.name?.value) {
+        const length = node.name.value.length / 2;
+
+        // convert the bytes32 (data) value to a Uint8Array
+        const bytes = new Uint8Array(length);
+        for (let i = 0; i < length; i++) {
+            bytes[i] = parseInt(node.name.value.slice(i * 2, i * 2 + 2), 16);
+        }
+
+        // decode the Uint8Array as a string and trim trailing null characters
+        const name = decoder.decode(bytes);
+        const trimmedName = name.replace(/\0/g, '');
+        return trimmedName;
```

Comment:
I think there's an ethers helper for this... we already have ethers as a dep so `import {ethers} from 'ethers'` and something like:

```suggestion
            return ethers.decodeBytes32String(node.name.value);
```

Reply:
much cleaner, thanks!